src/test/regress/sql/multi_fix_partition_shard_index_names.sql

```diff
@@ -0,0 +1,99 @@
+CREATE TABLE p PARTITION OF dist_partitioned_table FOR VALUES FROM ('2019-01-01') TO ('2020-01-01');
+
+-- create an index on parent table
+-- we will see that it doesn't matter whether we name the index on parent or not
+-- indexes auto-generated on partitions will not use this name
+CREATE INDEX short ON dist_partitioned_table USING btree (another_col, partition_col);
+
+SELECT tablename, indexname FROM pg_indexes WHERE schemaname = 'fix_idx_names' ORDER BY 1, 2;
+
+\c - - - :worker_1_port
+-- Note that, the shell table from above partition_table_with_very_long_name
+-- and its shard partition_table_with_very_long_name_910008
+-- have the same index name: partition_table_with_very_long_na_another_col_partition_col_idx
+SELECT tablename, indexname FROM pg_indexes WHERE schemaname = 'fix_idx_names' ORDER BY 1, 2;
+
+\c - - - :master_port
+-- this should fail because of the name clash explained above
+SELECT start_metadata_sync_to_node('localhost', :worker_1_port);
+
+-- let's fix the problematic table
+SET search_path TO fix_idx_names, public;
+SELECT fix_partition_shard_index_names('dist_partitioned_table'::regclass);
+
+\c - - - :worker_1_port
+-- shard id has been appended to all index names which didn't end in shard id
+-- even though partition p wasn't problematic since the shard id was inside the name
+-- and not truncated, we still append the shard id to the end of the name
+-- this goes in line with Citus's way of naming indexes of shards: always append shardid to the end
+SELECT tablename, indexname FROM pg_indexes WHERE schemaname = 'fix_idx_names' ORDER BY 1, 2;
+
+\c - - - :master_port
+SET search_path TO fix_idx_names, public;
+
+-- this should now work
+SELECT start_metadata_sync_to_node('localhost', :worker_1_port);
+
+-- if we run this command again, the names will not change anymore since shardid is appended to them
+SELECT fix_partition_shard_index_names('dist_partitioned_table'::regclass);
+SELECT fix_all_partition_shard_index_names();
+
+\c - - - :worker_1_port
+SELECT tablename, indexname FROM pg_indexes WHERE schemaname = 'fix_idx_names' ORDER BY 1, 2;
+
+\c - - - :master_port
+SET search_path TO fix_idx_names, public;
+SET citus.shard_replication_factor TO 1;
+SET citus.next_shard_id TO 910020;
+
+-- if we explicitly create index on partition-to-be table, Citus handles the naming
+-- hence we would have no broken index names
+CREATE TABLE another_partition_table_with_very_long_name (dist_col int, another_col int, partition_col timestamp);
+SELECT create_distributed_table('another_partition_table_with_very_long_name', 'dist_col');
+CREATE INDEX ON another_partition_table_with_very_long_name USING btree (another_col, partition_col);
+ALTER TABLE dist_partitioned_table ATTACH PARTITION another_partition_table_with_very_long_name FOR VALUES FROM ('2020-01-01') TO ('2021-01-01');
+SELECT tablename, indexname FROM pg_indexes WHERE schemaname = 'fix_idx_names' ORDER BY 1, 2;
+
```

Comment:
Can we add a test with weird index name like CREATE INDEX "weird index name !!" ON table; to make sure that it works as expected.